src/libraries/System.Private.CoreLib/src/System/Threading/WaitHandle.Windows.cs

```diff
@@ -102,8 +131,36 @@ private static int SignalAndWaitCore(IntPtr handleToSignal, IntPtr handleToWaitO
         {
             Debug.Assert(millisecondsTimeout >= -1);
 
-            int ret = (int)Interop.Kernel32.SignalObjectAndWait(handleToSignal, handleToWaitOn, (uint)millisecondsTimeout, Interop.BOOL.FALSE);
+            ulong startTime = 0;
+            if (millisecondsTimeout != -1)
+            {
+                startTime = Interop.Kernel32.GetTickCount64();
```

Comment:
Could you please use `Environment.TickCount64` and `long` for the types of local variable? It is going to make ulong cast in `(ulong)millisecondsTimeout` unnecessary.